Well my CMake version is 4.2.2, using the Kitware apt repos for it, and I'm running Debian 13 on the build machine.
I'm trying to build this for Debug mode, and my gcc version is 14.2.0.
This is running over SSH connected to a local Linux machine with VSCode.

I checked and the tmj.sym file is in the src folder.

Here is my CMakeLists.txt that I am using for this, I have mostly just used the example from the readme for running this.
```
cmake_minimum_required(VERSION 3.22)

set(PROJECT_NAME "c-tiled-test")

# Get most source files
file(GLOB MAIN_SOURCES "src/*.c")

project(${PROJECT_NAME} C)

# Set the C standard
set(CMAKE_C_STANDARD 11)
set(CMAKE_C_STANDARD_REQUIRED ON)

include(FetchContent)

# Disable update cache for this, makes the project work offline
set(FETCHCONTENT_UPDATES_DISCONNECTED ON)

#-------------
# Libtmj
# Changed to use my fork, I had to modify the .gitmodules to make this part work.
#-------------

FetchContent_Declare(libtmj
#   GIT_REPOSITORY https://github.com/Zer0-One/libtmj.git
  GIT_REPOSITORY https://github.com/kelson8/libtmj.git
  GIT_TAG        e4f38d478a2f217f982701380485a3136d399e6f
)
FetchContent_MakeAvailable(libtmj)

#-------------
# Jansson for Json
#-------------

FetchContent_Declare(jansson
  GIT_REPOSITORY https://github.com/akheron/jansson.git
  GIT_TAG        v2.15.0
)
FetchContent_MakeAvailable(jansson)

add_executable(${PROJECT_NAME}
    ${MAIN_SOURCES}
    )

target_include_directories(${PROJECT_NAME} PRIVATE 
    src)

target_link_libraries(${PROJECT_NAME} PRIVATE
  tmj
  jansson
    )
```

And the errors from the linker when using this option

```
/usr/bin/ld: cannot open linker script file src/tmj.sym: No such file or directory
collect2: error: ld returned 1 exit status
gmake[2]: *** [_deps/libtmj-build/CMakeFiles/tmj.dir/build.make:167: _deps/libtmj-build/lib/libtmj.so.1.5.0] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:232: _deps/libtmj-build/CMakeFiles/tmj.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
```

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "test/Unity"]
 	path = test/Unity
-	url = git@github.com:ThrowTheSwitch/Unity.git
+	url = https://github.com/ThrowTheSwitch/Unity.git

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,9 +55,10 @@ target_compile_definitions(tmj PUBLIC
     LIBTMJ_VERSION="${PROJECT_VERSION}"
 )
 
+
 # Control symbol visibility
 if(NOT WIN32 AND NOT APPLE)
-    target_link_options(tmj PRIVATE "LINKER:--version-script=src/tmj.sym")
+    target_link_options(tmj PRIVATE "LINKER:--version-script=${CMAKE_CURRENT_SOURCE_DIR}/src/tmj.sym")
 endif()
 
 # Set include directories

